### PR TITLE
`Min`/`MaxSupportedVersion` and `IsSupportedVersion` on `PythonEngine`

### PR DESCRIPTION
**Pull request by lostmsu**
### Does this close any currently open issues?

fixes https://github.com/pythonnet/pythonnet/discussions/1724

### Checklist

Check all those that are applicable and complete.

-   [x] Make sure to include one or more tests for your change
-   [x] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by lostmsu** [comment]
:/ I am puzzled. What does this change have to do with segfaults? Only on *nix? Ugh...

File: src/runtime/PythonEngine.cs [hunk 1]
```diff
@@ -127,6 +127,10 @@ public static string PythonPath
             }
         }
 
+        public static Version MinSupportedVersion => new(3, 7);
```
Comment (filmor) [inline src/runtime/PythonEngine.cs:130]: `readonly`?

File: src/runtime/PythonEngine.cs [hunk 2]
```diff
@@ -127,6 +127,10 @@ public static string PythonPath
...
+        public static Version MaxSupportedVersion => new(3, 10, int.MaxValue, int.MaxValue);
```
Comment (filmor) [inline src/runtime/PythonEngine.cs:131]: Maybe something more "readable", like 999 instead of maxint?

Comment (lostmsu) [reply src/runtime/PythonEngine.cs:131]: I thought of that, or maybe do `MaxSupportedVersionExclusive`. Alternatively, we can only leave the `IsSupportedVersion` check and provide an error string in a separate property.

**Comment by filmor** [comment]
> :/ I am puzzled. What does this change have to do with segfaults? Only on *nix? Ugh...

Just tested it as well, it only happens on Mono...

**Comment by filmor** [comment]
Further observations:

- The culprit is the call, the comparisons in the tests seem fine
- Only the `enum` conversions with max int/long/ulong fail, if I remove these, the tests run though

**Pull request merged**